addons/mass_mailing/static/src/js/tours/mass_mailing_editor_tour.js

```diff
@@ -10,16 +10,41 @@ odoo.define('mass_mailing.mass_mailing_editor_tour', function (require) {
         trigger: '.o_app[data-menu-xmlid="mass_mailing.mass_mailing_menu_root"]',
     }, {
         trigger: 'button.o_list_button_add',
-    }, {
-        trigger: 'input[name="subject"]',
-        run: 'text Test',
     }, {
         trigger: 'div[name="contact_list_ids"] .o_input_dropdown > input[type="text"]',
     }, {
         trigger: 'li.ui-menu-item',
     }, {
-        trigger: 'iframe #default',
-        run: () => $('iframe').contents().find('o_editable').html('<p><br/></p>'),
+        trigger: '[name="body_arch"] iframe #empty',
+    }, {
+        trigger: '[name="body_arch"] iframe .o_editable'
+    }, {
+        trigger: '[name="body_arch"] iframe #email_designer_default_body [name="Title"] .ui-draggable-handle',
+        run: function (actions) {
+            actions.drag_and_drop('[name="body_arch"] iframe .o_editable', this.$anchor);
+        }
+    }, {
+        trigger: '[name="body_arch"] iframe .o_editable h1'
+    }, {
+        trigger: 'button.o_form_button_save',
+    }, {
+        trigger: 'label.o_field_invalid',
+    }, {
+        trigger: '[name="body_arch"] iframe .o_editable h1'
+    }, {
+        trigger: '[name="body_arch"] iframe .odoo-editor-editable',
+        run: function () {
+            const ownerDocument = this.$anchor[0].ownerDocument;
+            const paragraph = ownerDocument.createElement('p');
+            const linebreak = ownerDocument.createElement('br');
+            paragraph.append(linebreak);
+            const fragment = new DocumentFragment();
+            fragment.append(paragraph);
+            this.$anchor[0].replaceChildren(...fragment.childNodes);
+        }
```

Comment:
Explanations would be especially needed here.
Seems like a lot of manual DOM manipulations, is it really required?

Reply:
It is the verbose way to avoid using `.html("<p><br/></p>")` which would trigger the security flag, but maybe we don't care since it's a tour ?

And the justification for the step is that because we chose a "drag&drop" "template" to edit the mailing, the `body_arch` now contains a lot of preparatory nodes, and to avoid modifying the asserts in the python test which calls this tour, I replaced these nodes by the default value of `odoo-editor-editable` (an empty paragraph)

Reply:
Still don't think this is a good idea to do that many manual DOM manipulations.
This essentially forces tags into the editor instead of simulating user inputs (clicks, text, ...), which is what a tour is supposed to do.

Either you can easily "reset" the editor by typing something into it using a default "text Blabla" step.
Either you should create a second (small) tour that tests your specific use case.

Reply:
I'll remove this step and change the python asserts. They weren't particularly well chosen for the intended purpose anyway (prone to fail if the style is tweaked in the future)